editor-extensions/vscode/syntaxes/grain.json

```diff
@@ -714,6 +716,22 @@
         }
       ]
     },
+    "infinity": {
+      "patterns": [
+        {
+          "match": "\\b(Infinity)\\b",
+          "name": "constant.language.grain"
+        }
+      ]
+    },
+    "nan": {
+      "patterns": [
+        {
+          "match": "\\b(NaN)\\b",
+          "name": "constant.language.grain"
+        }
+      ]
+    },
     "number": {
```

Comment:
You'll want to add these to number instead of their own section, and they should be `constant.numeric.grain`.